### DISCUSSION TITLE
The best way to back up SFP stick?

### DISCUSSION THREAD

**Discussion by tdmadam** [Anime4000/RTL960x#28] [General]
Compared to other modules, V2801F and DFP have many partitions. When I bought V2801F recently the first thing I did was backup, especially "envinronment" mtd1,mtd2 and "config" mtd3 wher ELAN_MAC, VS_AUTH, MAC_KEY are factory saved. If SFP stick is damaged or mtd3 wiped then it will lose this factory data unless we have a backup.

```
# cat /proc/mtd 
dev:    size   erasesize  name
mtd0: 00040000 00001000 "boot"
mtd1: 00002000 00001000 "env"
mtd2: 00002000 00001000 "env2"
mtd3: 0003c000 00001000 "config"
mtd4: 0014c000 00001000 "k0"
mtd5: 00274000 00001000 "r0"
mtd6: 0014c000 00001000 "k1"
mtd7: 00274000 00001000 "r1"
mtd8: 00001000 00001000 "Partition_008"
mtd9: 00001000 00001000 "Partition_009"
mtd10: 00001000 00001000 "Partition_010"
mtd11: 00001000 00001000 "Partition_011"
mtd12: 0014c000 00001000 "linux"
mtd13: 00274000 00001000 "rootfs
```

On Linux I use these commands

```
# Listen on port 9999
Ubuntu# nc -l -p 9999 > mtd3

# Send mtd3 to Ubuntu
V2801F# cat /dev/mtd3 | nc 192.168.1.150 9999
```

With so many partitions, this is a bit of work. Can you recommend something else?

**Comment by Anime4000** [comment]
![image](https://user-images.githubusercontent.com/1908715/165987126-f9d19497-70bf-4c71-bd51-4cdf6274285e.png)
I put `sfp-sfpplus1` under WAN Interface Group so I can access `192.168.1.1` via NAT
doing `nc` stick to pc need Port Forward 😄

**Comment by tdmadam** [comment]
This post is informational. I wanted to document which files are deleted after using flash_erasell /dev/mtd3 on my two sticks, and therefore how important it is to back up the "config" partition and text output of `flash all` command.

**V2801F**
```
# BEFORE
# cd /var/config/
# ls -l
-rwxr-xr-x    1 admin    0             417 Jan  1 00:00 igmp_gpon.conf
-rw-r--r--    1 admin    0            7998 Jul 14  1970 lastgood.dong
-rw-r--r--    1 admin    0            7951 Jan  1 00:00 lastgood.xml
-rw-r--r--    1 admin    0            4100 Jan  1 00:00 lastgood.xml.check_failed
-rw-r--r--    1 admin    0            7951 Jan  1 00:00 lastgood.xml_bak
-rw-r--r--    1 admin    0            3247 Jan  1 00:00 lastgood_hs.xml
-rw-r--r--    1 admin    0               0 Jan  1 00:00 lastgood_hs.xml.check_failed
-rw-r--r--    1 admin    0            3247 Jan  1 00:00 lastgood_hs.xml_bak
-rw-r--r--    1 admin    0            3250 Jan  1 00:00 lastgood_mp_hs2.xml

# AFTER
# cd /var/config/
# ls -l
-rwxr-xr-x    1 admin    0             417 Jan  1 00:00 igmp_gpon.conf
-rw-r--r--    1 admin    0            7953 Jan  1 00:00 lastgood.xml
-rw-r--r--    1 admin    0            7953 Jan  1 00:00 lastgood.xml_bak
-rw-r--r--    1 admin    0            3247 Jan  1 00:00 lastgood_hs.xml
-rw-r--r--    1 admin    0            3247 Jan  1 00:00 lastgood_hs.xml_bak
```
</br>

**DFP-34X-2C2**
```
# BEFORE
# cd /var/config/
# ls -l
-rw-r--r--    1 admin    0           17646 Jan  1  1970 captcha.gif
-rw-r--r--    1 admin    0            1552 Jan  1 00:00 europa.data
-rw-r--r--    1 admin    0             763 Jan  1 00:00 igmp_gpon.conf
-rw-r--r--    1 admin    0            6134 Jan  1  1970 lastgood.xml
-rw-r--r--    1 admin    0            6143 Jan  1 00:05 lastgood.xml_bak
-rw-r--r--    1 admin    0            3342 Jan  1 00:05 lastgood_hs.xml
-rw-r--r--    1 admin    0            3157 Jan  1 00:00 lastgood_hs.xml.check_failed
-rw-r--r--    1 admin    0            3342 Jan  1 00:05 lastgood_hs.xml_bak
-rw-r--r--    1 admin    0            3213 Jan  1 00:00 lastgood_mp_hs2.xml
-rwxrwxrwx    1 admin    0             431 Dec 21  2021 omci_custom_opt.conf
-rwxrwxrwx    1 admin    0            7920 Dec 21  2021 omci_mib.cfg
-rw-r--r--    1 admin    0            1566 Jan  1 00:00 rtl8290b.dat

# AFTER
# cd /var/config/
# ls -l
-rwxr-xr-x    1 admin    0             763 Jan  1 00:00 igmp_gpon.conf
-rw-r--r--    1 admin    0            6144 Jan  1 00:00 lastgood.xml
-rw-r--r--    1 admin    0            3290 Jan  1 00:00 lastgood_hs.xml
-rw-r--r--    1 admin    0            3290 Jan  1 00:00 lastgood_mp_hs2.xml
-rwxrwxrwx    1 admin    0             431 Dec 21  2021 omci_custom_opt.conf
-rwxrwxrwx    1 admin    0            7920 Dec 21  2021 omci_mib.cfg
```

**Reply by SMSegura** [reply]
Hello.
What would be the procedure to restore /dev/mtd3 ?
Thanks for any information

**Reply by Anime4000** [reply]
when mtd3 is blank, stick will fill with default settings

**Comment by hugomon** [comment]
I have a problem with my Ufiber UF-Instant, the netcat command is not available, so I can't backup or upload anything

```
# nc
/bin/sh: nc: not found
```

Maybe the reason is an old version of the stick's firmware?

```
# nv getenv | grep sw_ver
sw_version0=v4.1.0.359-b87.191101.0940
sw_version1=v4.1.0.359-b87.191101.0940
```

PS: also I can only connect via telnet, the ssh connection is always refused.

**Reply by hugomon** [reply]
Well I solved using scp pull from a linux machine (scp push was not working from the stick):

`scp ubnt@192.168.1.1:/tmp/mtd13 /home/user/backup/mtd13`

Files were previously created at the `/tmp` folder with `cat /dev/mtd13 > /tmp/mtd13`

**Reply by Anime4000** [reply]
this very clean setup, single command-line

**Reply by collerok** [reply]
also had this problem that NC does not exist, solved it by upgrading firmware by placing it to /tmp and then  run `/bin/fwupdate /tmp/fw.bin`
device updated itself and NC become working
**WARNING**: Do NOT update fw to 'v4.4.2.1248' this version block mtd7 from being overwritten by nc or flashall (set mtd7 as readonly)

**Reply by amaduain** [reply]
@collerok 
> also had this problem that NC does not exist, solved it by upgrading firmware by placing it to /tmp and then run `/bin/fwupdate /tmp/fw.bin` device updated itself and NC become working **WARNING**: Do NOT update fw to 'v4.4.2.1248' this version block mtd7 from being overwritten by nc or flashall (set mtd7 as readonly)

Can you give me more detail? I downloaded an img file from [here](https://github.com/stich86/UF-Instant-Mod/tree/main/Original%20Firmwares/v4.3.1.913-d48.210415.0811)
It is enough to change the extension? (I guess it is not :-()

**Reply by amaduain** [reply]
> @collerok
> 
> > also had this problem that NC does not exist, solved it by upgrading firmware by placing it to /tmp and then run `/bin/fwupdate /tmp/fw.bin` device updated itself and NC become working **WARNING**: Do NOT update fw to 'v4.4.2.1248' this version block mtd7 from being overwritten by nc or flashall (set mtd7 as readonly)
> 
> Can you give me more detail? I downloaded an img file from [here](https://github.com/stich86/UF-Instant-Mod/tree/main/Original%20Firmwares/v4.3.1.913-d48.210415.0811) It is enough to change the extension? (I guess it is not :-()

Well it worked using the img file directly:
```
# /bin/fwupdate /tmp/UF-ONU.UF-INSTANT.v4.3.1.913-d48.210415.0811.img
Starting upgrade of image 1 with image /tmp/UF-ONU.UF-INSTANT.v4.3.1.913-d48.210415.0811.img...
Check firmware...
Magic:    UBNT.ONU

```

**Comment by ghost** [comment]
> Compared to other modules, V2801F and DFP have many partitions. When I bought V2801F recently the first thing I did was backup, especially "envinronment" mtd1,mtd2 and "config" mtd3 wher ELAN_MAC, VS_AUTH, MAC_KEY are factory saved. If SFP stick is damaged or mtd3 wiped then it will lose this factory data unless we have a backup.
> 
> ```
> # cat /proc/mtd 
> dev:    size   erasesize  name
> mtd0: 00040000 00001000 "boot"
> mtd1: 00002000 00001000 "env"
> mtd2: 00002000 00001000 "env2"
> mtd3: 0003c000 00001000 "config"
> mtd4: 0014c000 00001000 "k0"
> mtd5: 00274000 00001000 "r0"
> mtd6: 0014c000 00001000 "k1"
> mtd7: 00274000 00001000 "r1"
> mtd8: 00001000 00001000 "Partition_008"
> mtd9: 00001000 00001000 "Partition_009"
> mtd10: 00001000 00001000 "Partition_010"
> mtd11: 00001000 00001000 "Partition_011"
> mtd12: 0014c000 00001000 "linux"
> mtd13: 00274000 00001000 "rootfs
> ```
> 
> On Linux I use these commands
> 
> ```
> # Listen on port 9999
> Ubuntu# nc -l -p 9999 > mtd3
> 
> # Send mtd3 to Ubuntu
> V2801F# cat /dev/mtd3 | nc 192.168.1.150 9999
> ```
> 
> With so many partitions, this is a bit of work. Can you recommend something else?

Hi, how to transfer mtd from Ubuntu to sfp stick? Please help :(

**Comment by dinamicus** [comment]
hello folks can someone share files that i've lost after flash_eraseall /dev/mtd3 . sad story
these ones please if it's possible. for DFP-34X-2C2 for sure

europa.data
rtl8290b.dat

**Reply by dinamicus** [reply]
ok nwm i've found a script to restore such files ) here /etc/scripts/insert_europa.sh

**Comment by Thaad-QR** [comment]
what is command to backup and resort please?

**Reply by Anime4000** [reply]
Backup
```
# Listen on port 9999
Ubuntu# nc -l -p 9999 > mtd3

# Send mtd3 to Ubuntu
V2801F# cat /dev/mtd3 | nc 192.168.1.150 9999
```

Restore
```
# Listen on port 9999
V2801F# nc -l -p 9999 > /dev/mtd3

# Send mtd3 to V2801F
Ubuntu# cat mtd3 | nc 192.168.1.150 9999
```

just like that

**Reply by bademux** [reply]
Hi,
Looks like on Openwrt it is not that easy, ONT is before wan. 
Is there generic way to do it in reverse way, by connecting to the ONT, some FTP server?
